lib/dependabot/update_checkers/java_script/npm_and_yarn/version_resolver.rb

```diff
@@ -318,13 +318,19 @@ def newly_broken_peer_reqs_from_dep
               select { |dep| dep[:requiring_dep_name] == dependency.name }
           end
 
+          def lockfiles_for_path(lockfiles:, path:)
+            lockfiles.select do |lockfile|
+              File.dirname(lockfile.name) == File.dirname(path)
+            end
+          end
+
           def run_checker(path:, version:)
-            if [*package_locks, *shrinkwraps].any?
-              run_npm_checker(path: path, version: version)
+            if lockfiles_for_path(lockfiles: [*package_locks, *shrinkwraps],
+                                  path: path).any?
+              return run_npm_checker(path: path, version: version)
             end
 
-            run_yarn_checker(path: path, version: version) if yarn_locks.any?
-            run_yarn_checker(path: path, version: version) if lockfiles.none?
+            run_yarn_checker(path: path, version: version)
```

Comment:
Doesn't this mean you'll run the yarn checker even if there are only npm lockfiles?

Reply:
Changed to return if there are any npm lockfiles so won't run both yarn and npm if both exist, but maybe better to only run yarn if both npm and yarn lockfiles exist? I *think* both give the same errors based what's in package.json

Reply:
Ah, yeah, good point for the update checker I guess we can get away with only running one. Worth putting in a comment explaining that?